compiler/src/dotty/tools/dotc/transform/init/Errors.scala

```diff
@@ -48,13 +48,6 @@ object Errors {
       }
       sb.toString
     }
-
-    /** Flatten UnsafePromotion errors
-     */
-    def flatten: Errors = this match {
-      case unsafe: UnsafePromotion => unsafe.errors.flatMap(_.flatten)
-      case _ => this :: Nil
-    }
```

Comment:
We may still need this, in case the _nested_ error is another `UnsafePromotion`. But that's OK as well, so let's remove it for now.